pkg/translator/jaeger/jaegerproto_to_traces.go

```diff
@@ -252,30 +271,47 @@ func setInternalSpanStatus(attrs pdata.AttributeMap, dest pdata.SpanStatus) {
 	}
 }
 
-func getStatusCodeValFromAttr(attrVal pdata.AttributeValue) (int, error) {
-	var codeVal int64
+// extractStatusDescFromAttr returns the OTel status description from attrs
+// along with true if it is set. Otherwise, an empty string and false are
+// returned. The OTel status description attribute is deleted from attrs in
+// the process.
+func extractStatusDescFromAttr(attrs pdata.AttributeMap) (string, bool) {
+	if msgAttr, ok := attrs.Get(conventions.OtelStatusDescription); ok {
+		msg := msgAttr.StringVal()
+		attrs.Delete(conventions.OtelStatusDescription)
+		return msg, true
+	}
+	return "", false
+}
+
+// codeFromAttr returns the integer code value from attrVal. An error is
+// returned if the code is not represented by an integer or string value in
+// the attrVal or the value is outside the bounds of an int representation.
+func codeFromAttr(attrVal pdata.AttributeValue) (int, error) {
+	var val int
 	switch attrVal.Type() {
 	case pdata.AttributeValueTypeInt:
-		codeVal = attrVal.IntVal()
+		i := attrVal.IntVal()
+		if i > math.MaxInt32 || i < math.MinInt32 {
+			return 0, fmt.Errorf("%w: %d", errRange, val)
+		}
```

Comment:
Do we really care about this check? What value does it bring? Does all this complication comes from the fact that `tracetranslator.StatusCodeFromHTTP` accepts int?

Reply:
AFAIK, yes. It was there before and I didn't want to introduce an unreported truncation error by just casting to an `int`.

Reply:
From async conversation: let's update `tracetranslator.StatusCodeFromHTTP` to accept an `int64` in a separate PR an avoid this entirely.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6699